projects/debouncer/src/debouncer.c

```diff
@@ -0,0 +1,58 @@
+#include "debouncer.h"
+
+#include "log.h"
+#include "soft_timer.h"
+#include "stm32f0xx_interrupt.h"
+
+#define INTERRUPT_MASKING_DURATION_MS 50
+
+// This is the callback for the soft timer. Gets the state and compares with old state.
+// If states are equal, it means that we had a button input. So it runs the given callback.
+static void prv_debouncer_timer_callback(SoftTimerID timer_id, void *context) {
```

Comment:
In our naming scheme, we replace the module name with `prv`. For example, instead of `debouncer_timer_callback` we could just use `prv_timer_callback`.